### PR TITLE
ENH: IPM custom status

### LINKED ISSUE #673: Custom Status Print for IPM
## Expected Behavior
```
In[1]: ipm2
Out[1]:
XPP SB2 IPM: Target 2 IN [2.0 um Silicon Nitride]
Target Position: 32.015 [mm]
Wave 8 Diode Position (x, y): 0.013, 0.025 [mm]
```
Where exactly to get/store the 2.0 um Silicon Nitride information is an open discussion item.

## Context

https://docs.google.com/document/d/1dmalnT4xLPWoJESsMHTxzIyzlGCBgTQjGNYwDhU-pzo/edit#heading=h.ylcn31lb55xr

### PR DESCRIPTION
**Pull request by cristinasewell**
Override the status info handler. 
 - [ ] I still need to get the target composition.
 - [ ] I need to double check if what I'm printing is the correct/wanted info.

## Motivation and Context

Trying to implement and close #673 

## How Has This Been Tested?

```
In [5]: i
Out[5]:
XPP SB2 IPM: Target 5 OUT
Target Position: 0.0 [mm]
IPIMB Diode Position(x, y): 0.1063, -45.9985 [mm]
```
## Where Has This Been Documented?

Docstring

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive docstrings, including context and API
- [ ] New/changed functions and methods are covered in the test suite where possible
- [ ] Test suite passes locally
- [ ] Test suite passes on travis
- [ ] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [ ] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: pcdsdevices/ipm.py [hunk 1]
```diff
@@ -102,6 +105,48 @@ class IPMMotion(BaseInterface, Device):
...
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_pos = status_info.get('target', {}).get('motor', {}).get(
+                                     'position', 'N/A')
+        t_units = status_info.get('target', {}).get('motor', {}).get(
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_state_num = status_info.get('target', {}).get('state', {}).get(
+                                           'value', 'N/A')
+        target_state = status_info.get('target', {}).get('position', 'N/A')
+
+        name = f'{name}: Target {target_state_num} {target_state}'
```
Comment (klauer) [inline pcdsdevices/ipm.py:142]: If I were you, I'd consider using the following pattern:
```suggestion
        return f"""\
{name}: Target {target_state_num} {target_state}
Target Position: {target_pos:.4f} [{t_units}]
Wave 8 Diode Position(x, y):
{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
"""
```

(wouldn't recommend "accepting" the suggestion, as I wrote this in the comment text box and not a proper editor)

Comment (cristinasewell) [reply pcdsdevices/ipm.py]: @klauer  thanks for the suggestions, is this what you meant 0efcae1 ?
This above prints:
```
In [3]: ipm
Out[3]:
XPP SB2 IPM: Target 5 OUT
Target Position: 0.0000 [mm]
Wave 8 Diode Position(x, y): 0.1063, -45.9985 [mm]
```
i will probably have to update the target position though, don't think i need a `.4f `there

Comment (klauer) [reply pcdsdevices/ipm.py]: Yep, looks good. Purely a simplification/stylistic change.

Comment (cristinasewell) [reply pcdsdevices/ipm.py]: ok thanks, i'll change it for the others too

File: pcdsdevices/ipm.py [hunk 2]
```diff
@@ -102,6 +105,46 @@ class IPMMotion(BaseInterface, Device):
...
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_state_num = status_info.get('target', {}).get('state', {}).get(
+                                           'value', 'N/A')
+        target_state = status_info.get('target', {}).get('position', 'N/A')
+
+        return f"""\
+{name}: Target {target_state_num} {target_state}
+Target Position: {target_pos} [{t_units}]
+Wave 8 Diode Position(x, y): \
+{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
```
Comment (ZryletTC) [inline pcdsdevices/ipm.py:145]: Not all IPMs use Wave8s. Can we add something like this?
```suggestion
        if 'ipimb' in status_info.keys():
            diode_type = 'IPIMB '
        elif 'wave8' in status_info.keys():
            diode_type = 'Wave8 '
        else:
            diode_type = ''

        return f"""\
{name}: Target {target_state_num} {target_state}
Target Position: {target_pos} [{t_units}]
{diode_type}Diode Position(x, y): \
{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
```

Comment (cristinasewell) [reply pcdsdevices/ipm.py]: Thanks for pointing this out @ZryletTC !

**Comment by cristinasewell** [comment]
Does anybody know where I can get the target composition from? 
```
[2.0 um Silicon Nitride]
```
as mentioned in #673 ?

**Comment by ZLLentz** [comment]
I'm going to quickly comment for future records that Silke proposed an answer to this in https://github.com/pcdshub/pcdsdevices/issues/673#issuecomment-726284808 and in the following conversation.

**Comment by cristinasewell** [comment]
I am not sure if the implementation in a3a706b is what we are after for the target composition - also, I did not test it with real composition information so we might have to wait and test it before merging this.

**Comment by ZLLentz** [comment]
I think that implementation is the most straightforward way to do it.

**Comment by ZLLentz** [comment]
I disagree on waiting- I think we merge, get feedback, and revisit if there's a small issue here.

**Comment by cristinasewell** [comment]
> I disagree on waiting- I think we merge, get feedback, and revisit if there's a small issue here. Otherwise this gets stuck forever. If we deploy this we get to say "the composition information will appear if you fill in the PVs" creating a reason to fill them in.

Sounds good @ZLLentz

**Comment by ZLLentz** [comment]
can I mark ready for review and merge?

**Comment by cristinasewell** [comment]
> can I mark ready for review and merge?

I always forget to mark ready for review - just did.

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**